src/components/CatalogOverlay/CatalogSubplotComponent/CatalogSubplotComponent.tsx

```diff
@@ -164,10 +228,19 @@ export class CatalogScatterComponent extends React.Component<WidgetProps> {
             const catalogFileId = this.widgetStore.catalogOverlayWidgetStore.catalogInfo.fileId;
             AppStore.Instance.updateCatalogProfiles(catalogFileId);
             let selectedPointIndices = [];
-            const points = event.points;
+            const points = event.points as any;
             for (let index = 0; index < points.length; index++) {
                 const selectedPoint = points[index];
-                selectedPointIndices.push(selectedPoint.pointIndex);
+                if (this.widgetStore.plotType === CatalogPlotType.D2Scatter) {
+                    selectedPointIndices.push(selectedPoint.pointIndex);   
```

Comment:
since we _know_ the length of `selectedPointIndices` beforehand, I think you should pre-allocate it and assign it instead of using `push`, because this can be much slower